### PR TITLE
MAG-349: Update order status

### PR DESCRIPTION
**Pull request by BenoitAlixAgenceDnD**
# ⚠️ Requirements
Reviewer, please take a look at those requirements before merging on `master` :

- [x] Check that plugin version has been upgrated and are identical in both `composer.json` and `Helper/Config.php` files

### CODE REVIEW

File: Helper/Data.php [hunk 1]
```diff
@@ -451,6 +458,19 @@ public function updateOrder($order)
...
+   * Get payment
+   *
+   * @param $order
+   * @param $storeId
+   * @return \Payplug\Resource\Payment
+   */
+    public function getPayment($order, $storeId){
+      $orderPayment = $this->getPaymentForOrder($order);
+      $payplugPayment = $orderPayment->retrieve($storeId);
+      return $payplugPayment;
```
Comment (TristanBouchierAgenceDnD) [inline Helper/Data.php:471]: Add a line jump before return

File: Controller/Payment/CheckPayment.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    $order = $this->salesOrderFactory->create();
+    $order->loadByIncrementId($lastIncrementId);
+
+    if (!$order->getId()) {
+      throw new \Exception(sprintf('Could not retrieve order with id %s', $lastIncrementId));
+    }
+
+    return $order;
+  }
+
```
Comment (TristanBouchierAgenceDnD) [inline Controller/Payment/CheckPayment.php:81]: Remove useless line jump

File: Controller/Payment/CheckPayment.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        "error" => true,
+        'message' => __('The transaction was aborted and your card has not been charged'),
+      ];
+      $response->setData($data);
+      return $response;
+    }
+
+    $data = ['error' => false];
+    $response->setData($data);
+    return $response;
```
Comment (TristanBouchierAgenceDnD) [inline Controller/Payment/CheckPayment.php:54]: Add line jump before return

File: Controller/Payment/CheckPayment.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    $payment = $this->payplugHelper->getPayment($order, $store_id);
+    if ((isset($payment->failure)) && (!empty($payment->failure)) || ($payment->is_paid === false && is_null($payment->paid_at))) {
+      $order->setStatus(Order::STATE_CANCELED);
+      $this->payplugHelper->updateOrder($order, ['status'=>Order::STATE_CANCELED]);
+      $data = [
+        "error" => true,
+        'message' => __('The transaction was aborted and your card has not been charged'),
+      ];
+      $response->setData($data);
+      return $response;
```
Comment (TristanBouchierAgenceDnD) [inline Controller/Payment/CheckPayment.php:49]: Add line jump before return

File: Controller/Payment/CheckPayment.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+
+  /**
+   * Retrieve PayPlug Standard payment url
+   *
+   * @return Json
+   */
+  public function execute()
+  {
+
```
Comment (TristanBouchierAgenceDnD) [inline Controller/Payment/CheckPayment.php:29]: Remove useless line jump

File: Controller/Payment/CheckPayment.php [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+use Payplug\Payments\Logger\Logger;
+
+use Magento\Sales\Model\OrderRepository;
+use Payplug\Payments\Exception\OrderAlreadyProcessingException;
+use Payplug\Payments\Model\Order\Payment;
+
+
+class CheckPayment extends AbstractPayment
+{
+
```
Comment (TristanBouchierAgenceDnD) [inline Controller/Payment/CheckPayment.php:20]: Remove those two useless line jumps

File: Controller/Payment/CheckPayment.php [hunk 7]
```diff
@@ -0,0 +1,82 @@
+<?php
+
```
Comment (TristanBouchierAgenceDnD) [inline Controller/Payment/CheckPayment.php:2]: Add declare(strict_types=1);

File: Controller/Payment/CheckPayment.php [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+namespace Payplug\Payments\Controller\Payment;
+
+use Magento\Framework\Controller\Result\Json;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Framework\Exception\PaymentException;
+use Magento\Sales\Model\Order;
+use Payplug\Exception\PayplugException;
+use Payplug\Payments\Helper\Data as Helper;
+use Payplug\Payments\Logger\Logger;
+
```
Comment (TristanBouchierAgenceDnD) [inline Controller/Payment/CheckPayment.php:12]: Remove useless line jump and sort the uses alphabetically

File: Controller/Payment/CheckPayment.php [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+  /**
+   * Get last order
+   *
+   * @return Order
+   *
+   * @throws \Exception
+   */
+  private function getLastOrder()
```
Comment (TristanBouchierAgenceDnD) [inline Controller/Payment/CheckPayment.php:65]: Add the return type "private function getLastOrder(): Order"

File: Controller/Payment/CheckPayment.php [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+use Magento\Framework\Exception\PaymentException;
+use Magento\Sales\Model\Order;
+use Payplug\Exception\PayplugException;
+use Payplug\Payments\Helper\Data as Helper;
+use Payplug\Payments\Logger\Logger;
+
+use Magento\Sales\Model\OrderRepository;
+use Payplug\Payments\Exception\OrderAlreadyProcessingException;
+use Payplug\Payments\Model\Order\Payment;
+
```
Comment (TristanBouchierAgenceDnD) [inline Controller/Payment/CheckPayment.php:16]: Only one line break between the uses and the class declaration

File: Helper/Data.php [hunk 11]
```diff
@@ -451,6 +458,19 @@ public function updateOrder($order)
...
     }
 
+  /**
+   * Get payment
+   *
+   * @param $order
+   * @param $storeId
+   * @return \Payplug\Resource\Payment
+   */
+    public function getPayment($order, $storeId){
```
Comment (TristanBouchierAgenceDnD) [inline Helper/Data.php:468]: The bracket must open on the next line

Comment (TristanBouchierAgenceDnD) [inline Helper/Data.php:468]: Return the type : "public function getPayment($order, $storeId): Payment"

File: Helper/Data.php [hunk 12]
```diff
@@ -451,6 +458,19 @@ public function updateOrder($order)
         return $order;
     }
 
+  /**
+   * Get payment
+   *
+   * @param $order
+   * @param $storeId
+   * @return \Payplug\Resource\Payment
```
Comment (TristanBouchierAgenceDnD) [inline Helper/Data.php:466]: Simplify FQN (Add "use Payplug\Resource\Payment" on the top of the class) and make Payment the @return

File: Helper/Data.php [hunk 13]
```diff
@@ -410,7 +410,7 @@ public function updateOrderStatus($order, $save = true)
      *
      * @throws OrderAlreadyProcessingException
      */
-    public function updateOrder($order)
+    public function updateOrder($order, $data = [])
```
Comment (TristanBouchierAgenceDnD) [inline Helper/Data.php:413]: Add the return type "public function updateOrder($order, $data = []): Order"

**Review by TristanBouchierAgenceDnD — CHANGES_REQUESTED** [review]
Mostly lines breaks and return types to fix on php classes. The other files are good to go.

**Review by TristanBouchierAgenceDnD — APPROVED** [review]

**Comment by TristanBouchierAgenceDnD** [comment]
OK for the backend, waiting for @antoine-fontaine 's approval on the frontend side to merge in QA.

**Review by antoine-fontaine — APPROVED** [review]

**Pull request merged**